temporalcli/commands.workflow.go

```diff
@@ -223,8 +224,45 @@ func (*TemporalWorkflowTraceCommand) run(*CommandContext, []string) error {
 	return fmt.Errorf("TODO")
 }
 
-func (*TemporalWorkflowUpdateCommand) run(*CommandContext, []string) error {
-	return fmt.Errorf("TODO")
+func (c *TemporalWorkflowUpdateCommand) run(cctx *CommandContext, args []string) error {
+	cl, err := c.Parent.ClientOptions.dialClient(cctx)
+	if err != nil {
+		return err
+	}
+	defer cl.Close()
+
+	// Get raw input
+	input, err := c.buildRawInput()
+	if err != nil {
+		return err
+	}
+
+	request := &client.UpdateWorkflowWithOptionsRequest{
+		WorkflowID:          c.WorkflowId,
+		RunID:               c.RunId,
+		UpdateName:          c.Name,
+		FirstExecutionRunID: c.FirstExecutionRunId,
+		Args:                input,
+	}
+
+	updateHandle, err := cl.UpdateWorkflowWithOptions(cctx, request)
+	if err != nil {
+		return fmt.Errorf("unable to update workflow: %w", err)
+	}
+
+	var valuePtr interface{}
```

Comment:
I think elsewhere I did:

```suggestion
	var valuePtr json.RawMessage
```

Though I think it's basically the same thing here (it'll just be a `map[string]any` for an object which is turned right back to JSON) so no worries.